src/Microsoft.DotNet.Cli.Utils/Telemetry.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class Telemetry
+    {
+        private class Variables
+        {
+            public static readonly string InstrumentationKey = "74cc1c9e-3e6e-4d05-b3fc-dde9101d0254";
+            private static readonly string Prefix = "DOTNET_CLI_TELEMETRY_";
+            public static readonly string Optout = Prefix + "OPTOUT";
+        }
+
+        private class Properties
+        {
+            public static readonly string OSVersion = "OS Version";
+            public static readonly string OSPlatform = "OS Platform";
+            public static readonly string RuntimeId = "Runtime Id";
+            public static readonly string ProductVersion = "Product Version";
+        }
+
+        private static bool _isInitialized = false;
+        private static TelemetryClient _client = null;
+
+        private static Dictionary<string, string> _commonProperties = null;
+        private static Dictionary<string, double> _commonMeasurements = null;
+
+        static Telemetry()
+        {
+            if (_isInitialized)
+                return;
+
+            bool Optout = Env.GetBool(Variables.Optout);
+
+            if (Optout)
+                return;
+
+            try
+            {
+                _client = new TelemetryClient();
+                _client.InstrumentationKey = Variables.InstrumentationKey;
+                _client.Context.Session.Id = Guid.NewGuid().ToString();
+
+                var runtimeEnvironment = PlatformServices.Default.Runtime;
+                _client.Context.Device.OperatingSystem = runtimeEnvironment.OperatingSystem;
+
+                _commonProperties = new Dictionary<string, string>();
+                _commonProperties.Add(Properties.OSVersion, runtimeEnvironment.OperatingSystemVersion);
+                _commonProperties.Add(Properties.OSPlatform, runtimeEnvironment.OperatingSystemPlatform.ToString());
+                _commonProperties.Add(Properties.RuntimeId, runtimeEnvironment.GetRuntimeIdentifier());
+                _commonProperties.Add(Properties.ProductVersion, Product.Version);
+                _commonMeasurements = new Dictionary<string, double>();
+
+                _isInitialized = true;
+            }
+            catch (Exception) { }
```

Comment:
@LakshanF, I agree with you that this should not fail and have a negative effect on what the end user is trying to achieve.  That being said, do you think it would make sense to Debug.Fail or something like that so that there is a better chance issues will be caught during development in case something is wrong?